### PR TITLE
Adds cp2k script

### PR DESCRIPTION
**Pull request by bananenpampe**
Together with @PicoCentauri we would like to add an example for batched CP2K calculations from a list of ase.Atoms.
The example automatically installs cp2k and runs a testcalculation on a small frame.

### CODE REVIEW

**Comment by bananenpampe** [comment]
This does not work because no CP2K actually gets installed throught the conda install.
Maybe this is due to the virtual instances github action uses? - can it be changed to another instance type?

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+Batch run of CP2K calculations
+==============================
+
+.. start-body
+
+This is an example how to perform single point calculations based on list of structures
+using `CP2K <https://www.cp2k.org>`_ using its `reftraj
+functionality <https://manual.cp2k.org/trunk/CP2K_INPUT/MOTION/MD/REFTRAJ.html>`_. The
+inputs are a set of structures in :download:`example.xyz` using the DFT parameters
+defined in :download:`reftraj_template.cp2k` importing basisset and pseudopotentials
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]:
```suggestion
defined in :download:`reftraj_template.cp2k` importing basis set and pseudopotentials
```

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+from the local CP2K install. The reference DFT paramaters are taken from `Cheng et al.
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]:
```suggestion
from the local CP2K installation. The reference DFT parameters are taken from `Cheng et al.
```

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+Ab initio thermodynamics of liquid and solid water 2019
+<https://www.pnas.org/doi/10.1073/pnas.1815117116>`_. Due to the small size of the test
+structure and convergence issues, we have decreased the size of the ``CUTOFF_RADIUS``
+from :math:`6.0\,\mathrm{Å}` to :math:`3.0\,\mathrm{Å}`. For actual production
+calculations adapt the template!
+
+To run this example we use a bare exectubale called with ``cp2k``. If you want to use
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]:
```suggestion
To run this example, we use a bare executable called with ``cp2k``. If you want to use
```

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    new_frames += frames_dft
+
+new_fname = f"{splitext(basename(write_to_file))[0]}_dft.xyz"
+ase.io.write(f"{project_directory}/{new_fname}", new_frames)
+
+# %%
+# Perform calculations using ASE calculator
+# =========================================
+# Above we performed the calculations using an external bash script. ASE also provides a
+# calculator class that we can use the perform the caclulations with pur input file
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]: pur?? what you mean?

**Review by DavideTisi — COMMENTED** [review]
seems very good only few typos

File: examples/cp2k_run_batch/setup_reftraj.py (hunk 4 above)
Comment (PicoCentauri) [reply examples/cp2k_run_batch/setup_reftraj.py]: our xD

Comment (DavideTisi) [reply examples/cp2k_run_batch/setup_reftraj.py]: 👍😂

Comment (Luthaf) [reply examples/cp2k_run_batch/setup_reftraj.py]: 🐈‍⬛

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+# %%
+# Perform calculations using ASE calculator
+# -----------------------------------------
+# Above we performed the calculations using an external bash script. ASE also provides a
+# calculator class that we can use the perform the caclulations with our input file
+# without a detour of writing files to disk.
+#
+# To use the ASE calculator together with a custom input script this requires some
+# adjsutements. First the name of the executable that has the exact name ``cp2k_shell``.
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]:
```suggestion
# adjustments. First the name of the executable that has the exact name ``cp2k_shell``.
```

File: examples/cp2k_run_batch/setup_reftraj.py [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+# containing the computed energies and forces of the sinulation.
+
+frames_full = ase.io.read("example.xyz", ":")
+project_name = "test_calcs"  # name of the global PROJECT
+project_directory = "production"
+write_to_file = "out.xyz"
+
+# %%
+# Below we show the initial configuration of two water molecules in a cubic box with a
+# sidelength of :math:`\approx 4\,\mathrm{Å}`.
```
Comment (DavideTisi) [inline examples/cp2k_run_batch/setup_reftraj.py]:
```suggestion
# side length of :math:`\approx 4\,\mathrm{Å}`.
```

**Review by DavideTisi — APPROVED** [review]
i found two more typos but everything is fine

**Pull request merged**